wgpu/gui/qt.py

```diff
@@ -141,9 +141,6 @@ def __init__(self, *args, **kwargs):
         self._request_draw_timer.setSingleShot(True)
         self._request_draw_timer.timeout.connect(self.update)
 
-        # Get the window id one time. For some reason this is needed
-        # to "activate" the canvas. Otherwise the viz is not shown if
-        # one does not provide canvas to request_adapter().
         self.get_window_id()
```

Comment:
I changed it to this to make it work locally:

```suggestion
        if self.parent() is not None:
            self.get_window_id()
```

Reply:
I moved it to the canvas, so it's not necessary here. In fact, it looked like (part of?) the problematic use-case was that the widget was instantiated without a parent and later attached to the gui.